lib-src/rpc2/rp2gen/crout.c

```diff
@@ -185,13 +188,220 @@ static rp2_bool legal_struct_fields[] = {
 	/* RPC2_Byte */		RP2_TRUE,
 	/* RPC2_String */		RP2_TRUE,	/* Untested */
 	/* RPC2_CountedBS */	RP2_TRUE, /* Untested */
-	/* RPC2_BoundedBS */	RP2_TRUE, 
+	/* RPC2_BoundedBS */	RP2_TRUE,
 	/* RPC2_BulkDescriptor */	RP2_FALSE,  /* Untested */
 	/* RPC2_EncryptionKey */	RP2_TRUE,   /* Untested */
 	/* RPC2_Struct */		RP2_TRUE,
 	/* RPC2_Enum */		RP2_TRUE,
 	/* RPC2_Double */		RP2_TRUE
 };
+void print_struct_func(RPC2_TYPE *t, FILE *where, FILE *hfile, char *name) {
+    if (t->tag != RPC2_STRUCT_TAG)
+        return;
+    fprintf(where, "int pack_struct_%s (BUFFER *buf, %s *ptr, RPC2_Integer who) {\n", name, name);
+    fprintf(hfile, "int pack_struct_%s (BUFFER *buf, %s *ptr, RPC2_Integer who);\n", name, name);
+
+    VAR **v;
+	for (v=t->fields.struct_fields; *v!=NIL; v++) {
+        if (!legal_struct_fields[(int32_t) (*v)->type->type->tag]) {
+			printf("RP2GEN: illegal type for RPC2_Struct field: %s\n", (*v)->name);
+			exit(1);
+		}
+		print_pack_var("ptr->", *v, where);
+	}
+    fputs("    return 0;\n", where);
+    fputs("}\n", where);
+    fprintf(where, "int unpack_struct_%s (BUFFER *buf, %s *ptr, RPC2_Integer who) {\n", name, name);
+    fprintf(hfile, "int unpack_struct_%s (BUFFER *buf, %s *ptr, RPC2_Integer who);\n", name, name);
+
+	for (v=t->fields.struct_fields; *v!=NIL; v++) {
+        if (!legal_struct_fields[(int32_t) (*v)->type->type->tag]) {
+			printf("RP2GEN: illegal type for RPC2_Struct field: %s\n", (*v)->name);
+			exit(1);
+		}
+		print_unpack_var("ptr->", *v, where);
```

Comment:
Minor, but lets use the `fprintf` calling style and move where to the first argument